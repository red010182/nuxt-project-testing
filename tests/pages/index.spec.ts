import {mount} from "@vue/test-utils";
import $ from 'jquery';
import Index from "../../src/pages/index.vue";
import { mockPermissions } from '../../dummies/mockPermissions';

describe("index spec", () => {
  const wrapper = mount(Index);
  const vm: any = wrapper.vm;
  it("should render proper", () => {
    // expect(wrapper.element).toMatchSnapshot();
  });

  it("should init properly", () => {
    expect(vm.inputCode).toBe('')
    expect(vm.result).toBe(null)
    const permissionRoot = vm.permissionsRoot
    expect(permissionRoot)
    expect(JSON.stringify(permissionRoot)).toBe(JSON.stringify({
      code: '',
      label: 'root',
      children: mockPermissions,
    }))
  })

  it('search no found', () => {
    vm.inputCode = '  '
    vm.search()
    expect(vm.result).toBe('No Result')

    vm.inputCode = ''
    vm.search()
    expect(vm.result).toBe('No Result')
  })

  it('search found', () => {
    vm.inputCode = 'code B-1'
    vm.search()
    expect(vm.result).toBe('label B-1')
  })

});
