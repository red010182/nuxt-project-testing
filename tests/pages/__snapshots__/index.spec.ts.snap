// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index spec should render proper 1`] = `
<div>
  <div>
    <span>
      Input permission code:
    </span>

    <input
      autofocus="autofocus"
    />

    <button>
      Search
    </button>
  </div>
</div>
`;
